src/common/io.py

```diff
@@ -1337,43 +1326,60 @@ def _get_name_dict(self):
 
         return name_dict
 
-    def _get_trajectory(self, data_index):
+    def _get_trajectory(self, data_index, start_index = 0, stop_index = None):
         if isinstance(self._data_2, dict):
             self._verify_file_data()
-            
+
             if data_index in self._data_2:
                 return self._data_2[data_index]
 
             file_position  = self._data_2_info["file_position"]
             sizeof_type    = self._data_2_info["sizeof_type"]
             nbr_points     = self._data_2_info["nbr_points"]
             nbr_variables  = self._data_2_info["nbr_variables"]
-            
-            self._data_2[data_index] = fmi_util.read_trajectory(encode(self._fname), data_index, file_position, sizeof_type, int(nbr_points), int(nbr_variables))
+
+            # Account for sub-sets of data
+            if start_index is not None and start_index > 0:
```

Comment:
Also in a few other places; do we need these constant checks if start_index might be None? The default is 0?

Reply:
Good point, that was an old artifact. I have removed it.